Common/Extensions.cs

```diff
@@ -1006,5 +1007,38 @@ public static string ToSafeString(this PyObject pyObject)
                 return value;
             }
         }
+
+        /// <summary>
+        /// Tries to convert a <see cref="PyObject"/> into a managed object
+        /// </summary>
+        /// <typeparam name="T">Target type of the resulting managed object</typeparam>
+        /// <param name="pyObject">PyObject to be converted</param>
+        /// <param name="result">Managed object </param>
+        /// <returns>True if successful conversion</returns>
+        public static bool TryConvert<T>(this PyObject pyObject, out T result)
+            where T : class
+        {
+            result = default(T);
+
+            if (pyObject == null)
+            {
+                return true;
+            }
+
+            using (Py.GIL())
+            {
+                try
+                {
+                    result = pyObject.AsManagedObject(typeof(T)) as T;
+                }
+                catch
+                {
+                    // Do not throw or log the exception.
+                    // Return false as an exception means that the conversion could not be made.
+                }
+            }
+
+            return result != default(T);
```

Comment:
In cases such as this, I think it's best to have an explicit `return true` and explicit `return false` statement.

Inside the `try` block we can add `return true` following the `result = pyObject.AsManagedObject(typeof(T)) as T;` line and then this line can become `return false`.

While it may not apply in this case (or often at all) -- I prefer to avoid using `==` or `!=` on unknown types since these operators can be overloaded ([see](https://github.com/QuantConnect/Lean/blob/master/Common/Symbol.cs#L387) the `Symbol` class which equates `Symbol.Empty` and `null`!).

Also, I think it makes the code clearer by including the `return true` and `return false` statements in particular points in the control flow, since that's what really determines success -- that we were able to execute `pyObject.AsManagedObject(typeof(T))` without throwing an exception and the `result != default(T)` is a proxy check for the success of the previous method call.